plugins/catalog-backend-module-unprocessed/api-report.md

```diff
@@ -15,6 +16,6 @@ export default catalogModuleUnprocessedEntities;
 export class UnprocessedEntitiesModule {
   constructor(database: Knex, router: Pick<HttpRouterService, 'use'>);
   // (undocumented)
-  registerRoutes(): void;
+  registerRoutes({ permissions }: { permissions: PermissionEvaluator }): void;
```

Comment:
This is a breaking change now, so it should be called out as such if we can't make it backwards compatible.

Reply:
Ah, breaking because it requires the end user to add permissions. I like the idea of making it backwards compatible.

Reply:
I'll do some searching on my own - do any examples exist today where implementing permissions in a plugin is optionally provided?

Reply:
Umm, DevTools ;)

Reply:
I would say that if you can, don't break existing APIs.
Should maybe also just not allow deletions if you're not passing in permissions, and return an error that guides people to how to fix it.

Reply:
I need help adding permissions without it being a breaking change, and I don't see how permissions are optional in DevTools. 😞 I have to pass in permissions as options to the router. Is the right decision to make the entire `RouterOptions` object optional? Users will still have to pass in an empty object if I make permissions optional, correct? That's a breaking change 🤔. 

```ts
interface RouterOptions = {
  permissions?: PermissionsEvaluator
}
```

Reply:
Sorry, permissions are optional as far as creating the rules for them in DevTools but the `PermissionsEvaluator` is required and I'm not seeing any good examples where it's not :(

Reply:
Our internal version of this isn't wrapped with permissions 🙃 we didn't see it as a significant enough threat since the table repopulates when there is an entity with a valid error. 

I don't see a way to ship without breaking change. Should I strip out permissions and 🚢, or make it a breaking change? 

cc: @alde

Reply:
I think a breaking change isn't always bad, people can choose not to upgrade